src/wallet/wallet.ts

```diff
@@ -103,170 +142,442 @@ class HathorWalletServiceWallet extends EventEmitter {
+      // Should we handle any error?
+      throw err;
+    }
+  }
+
+  /**
+   * Check if wallet is ready and throw error if not ready
+   *
+   * @memberof HathorWalletServiceWallet
+   * @inner
+   */
+  private checkWalletReady() {
+    if (!this.isReady()) {
+      throw new WalletError('Wallet not ready');
     }
   }
 
+  /**
+   * Method executed when wallet is ready
+   *
+   * @memberof HathorWalletServiceWallet
+   * @inner
+   */
+  private async onWalletReady() {
+    this.setState(walletState.READY);
+    await this.getNewAddresses();
+  }
+
   /**
    * Get all addresses of the wallet
    *
-   * @memberof HathorWallet
+   * @memberof HathorWalletServiceWallet
    * @inner
    */
-  async getAllAddresses(): Promise<string[]> {
-    const response = await walletApi.getAddresses(this.walletId!);
-    let addresses = [];
+  async getAllAddresses(): Promise<GetAddressesObject[]> {
+    this.checkWalletReady();
+    const response = await walletApi.getAddresses(this);
+    let addresses: GetAddressesObject[] = [];
     if (response.status === 200 && response.data.success === true) {
       addresses = response.data.addresses;
     } else {
-      // TODO What error should be handled here?
+      throw new WalletRequestError('Error getting wallet addresses.');
     }
     return addresses;
   }
 
+  private async getNewAddresses() {
+    this.checkWalletReady();
+    const response = await walletApi.getNewAddresses(this);
+    let addresses: AddressInfoObject[] = [];
+    if (response.status === 200 && response.data.success === true) {
+      addresses = response.data.addresses;
+      this.newAddresses = addresses;
+      this.indexToUse = 0;
+    } else {
+      throw new WalletRequestError('Error getting wallet addresses to use.');
+    }
+  }
+
   /**
    * Get the balance of the wallet for a specific token
    *
-   * @memberof HathorWallet
+   * @memberof HathorWalletServiceWallet
    * @inner
    */
-  async getBalance(token: string | null = null) {
-    const response = await walletApi.getBalances(this.walletId!, token);
-    let balance = null;
+  async getBalance(token: string | null = null): Promise<GetBalanceObject[]> {
+    this.checkWalletReady();
+    const response = await walletApi.getBalances(this, token);
+    let balance: GetBalanceObject[] = [];
     if (response.status === 200 && response.data.success === true) {
       balance = response.data.balances;
     } else {
-      // TODO What error should be handled here?
+      throw new WalletRequestError('Error getting wallet balance.');
     }
     return balance;
   }
 
   /**
    * Get the history of the wallet for a specific token
    *
-   * @memberof HathorWallet
+   * @memberof HathorWalletServiceWallet
    * @inner
    */
-  async getTxHistory(options: { token?: string } = {}) {
+  async getTxHistory(options: { token?: string } = {}): Promise<GetHistoryObject[]> {
     // TODO Add pagination parameters
```

Comment:
TODO?

Reply:
Yes. This PR was supposed to handle only the token actions integration. I had to also do the authorization methods, otherwise it would not work. So I decided not to add any more features and this TODO is already removed and implemented in the PR that matches the faces signatures.

Reply:
Ok. Same case as our other conversation.